vocabularies/Common.xml

```diff
@@ -1454,6 +1454,93 @@ If the request succeeds the response will contain a JSON body of `Content-Type:
         <Annotation Term="Core.Description" String="Base URL for WebSocket connections" />
       </Term>
 
+      <Term Name="RecommendationsFunction" Type="Common.QualifiedName" Nullable="false" AppliesTo="EntityType">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Qualified name of a function that computes AI-based recommendations for entities of the annotated type (see [this template](#Template_RecommendationsFunction))" />
+      </Term>
+      <Function Name="Template_RecommendationsFunction" IsBound="true">
+        <Annotation Term="Common.Experimental" />
+        <Annotation Term="Core.Description" String="Template for functions that compute AI-based recommendations for the bound entity and its related entities" />
+        <Annotation Term="Core.LongDescription">
+          <String>If the function is invoked repeatedly in a stateful session, it SHOULD NOT
+recompute recommendations whose predictors have not changed since the previous invocation.
```

Comment:
Is this an instruction for the implementors of a recommendation function, or is it the function's desired behavior to not return recommendations that haven't changed since the last stateful call?

Reply:
Both. But since this is an implementation detail, I'll remove the sentence.